cmd/wigToTrainingSet/wigToTrainingSet.go

```diff
@@ -0,0 +1,182 @@
+			} else {
+				lineToWrite = fmt.Sprintf("%s\t%s\t%g\n", currFa.Name, dna.BasesToString(currFa.Seq), i.Values[(start+start+s.WindowSize)/2]) //start+window / 2 is the midpoint of the window.
+			}
+			//now we shard the training example into either the testing, training, or validation set.
+			currRand = rand.Float64()
+			if currRand < s.TestingProp {
+				_, err = fmt.Fprintf(testOut, lineToWrite)
+				exception.PanicOnErr(err)
+			} else if currRand < s.TestingProp+s.ValidationProp {
+				_, err = fmt.Fprintf(validateOut, lineToWrite)
+				exception.PanicOnErr(err)
+			} else {
+				_, err = fmt.Fprintf(trainOut, lineToWrite)
+				exception.PanicOnErr(err)
+			}
+
+			if s.IncludeRevComp {
+				dna.ReverseComplement(currFa.Seq)
+				if s.LogTransform {
+					lineToWrite = fmt.Sprintf("%s\t%s\t%g\n", fmt.Sprintf("%s_rev", currFa.Name), dna.BasesToString(currFa.Seq), math.Log(i.Values[(start+start+s.WindowSize)/2]))
+				} else {
+					lineToWrite = fmt.Sprintf("%s\t%s\t%g\n", fmt.Sprintf("%s_rev", currFa.Name), dna.BasesToString(currFa.Seq), i.Values[(start+start+s.WindowSize)/2])
+				}
+				//we don't regenerate currRand, for and rev for same sequence both land in the same shard.
+				if currRand < s.TestingProp {
+					_, err = fmt.Fprintf(testOut, lineToWrite)
+					exception.PanicOnErr(err)
+				} else if currRand < s.TestingProp+s.ValidationProp {
+					_, err = fmt.Fprintf(validateOut, lineToWrite)
+					exception.PanicOnErr(err)
+				} else {
+					_, err = fmt.Fprintf(trainOut, lineToWrite)
+					exception.PanicOnErr(err)
+				}
+			}
+		}
+	}
+	err = trainOut.Close()
+	exception.PanicOnErr(err)
+	err = testOut.Close()
+	exception.PanicOnErr(err)
+	err = validateOut.Close()
+	exception.PanicOnErr(err)
+}
+
+func getChromIndex(genome []fasta.Fasta, chrom string) int {
+	for i := range genome {
+		if genome[i].Name == chrom {
+			return i
+		}
+	}
+	log.Fatalf("Wig chromosome name: %v not found in target genome.", chrom)
+	return -1
+}
+
+func usage() {
+	fmt.Print(
+		"wigToTrainingSet - Converts a wig to a training, validation, and testing set for the GenomeSequenceConvNet.\n" +
+			"Usage:\n" +
+			"wigToTrainingSet input.wig genome.fa train.txt validate.txt test.txt\n" +
+			"options:\n")
+	flag.PrintDefaults()
+}
+
+type Settings struct {
+	InWigFile      string
+	InFastaFile    string
+	TrainFile      string
+	ValidateFile   string
+	TestFile       string
+	WindowSize     int
+	Stride         int
+	ValidationProp float64
+	TestingProp    float64
+	SetSeed        int64
+	Missing        float64
+	LogTransform   bool
+	IncludeRevComp bool
+	NoHeader       bool
+}
+
+func main() {
+	var expectedNumArgs int = 5
+	var missing *float64 = flag.Float64("missing", -10, "Sets the missing value. Regions with this number as the wig value will be skipped.")
+	var windowSize *int = flag.Int("windowSize", 400, "Sets the windowSize for mapping sequence to scalars.")
```

Comment:
What do you think about forcing windowSize to be odd, so that we get a true midpoint?